File: guild/src/test/java/com/os/operando/guild/QuartetTest.java
package com.os.operando.guild;

import org.junit.Test;

import java.time.LocalDate;
import java.util.Calendar;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class QuartetTest {

    @Test
    public void test() {
        Calendar calendar = Calendar.getInstance();
        Quartet<String, Integer, Boolean, Calendar> quartet = Quartet.create("quartet", 0, false, calendar);
        assertThat(quartet.getFirst(), is("quartet"));
        assertThat(quartet.getSecond(), is(0));
        assertThat(quartet.getThird(), is(false));
        assertThat(quartet.getFourth(), is(calendar));
    }
}

File: guild-kotlin/src/test/kotlin/com/os/operando/guild/kt/QuartetTest.kt
package com.os.operando.guild.kt

import org.hamcrest.Matchers.`is`
import org.junit.Assert.*
import org.junit.Test

class QuartetTest {

    @Test
    fun test() {
        val quartet = 1 to 10L to true to "quartet"
        assertThat(quartet.first, `is`(1))
        assertThat(quartet.second, `is`(10L))
        assertThat(quartet.third, `is`(true))
        assertThat(quartet.fourth, `is`("quartet"))
    }

    @Test
    fun testToList() {
        val l = (0 to 1 to 2 to 3).toList()
        assertThat(l.size, `is`(4))
        assertThat(l[0], `is`(0))
        assertThat(l[1], `is`(1))
        assertThat(l[2], `is`(2))
        assertThat(l[3], `is`(3))
    }
}